pkg/provider/provider.go

```diff
@@ -123,3 +125,11 @@ func CompareHostOfURLS(uri1, uri2 string) bool {
 	}
 	return u1.Host == u2.Host
 }
+
+func ValidateYaml(content []byte, filename string) error {
+	var validYaml any
+	if err := yaml.Unmarshal(content, &validYaml); err != nil {
```

Comment:
@tricktron why did you chose "k8s.io/apimachinery/pkg/util/yaml" over "gopkg.in/yaml.v2"?, is there any specific reason?

Reply:
> @tricktron why did you choose "k8s.io/apimachinery/pkg/util/yaml" over "gopkg.in/yaml.v2"?, is there any specific reason?

@zakisk Ups, I just reused the `k8s.io/apimachinery/pkg/util/yaml` during the refactoring from [this pr](https://github.com/openshift-pipelines/pipelines-as-code/pull/1578/files#diff-5b6abe4f3dd00e71b0bce9400ca69023651eda401e8b9149734dbe65d79f66dcR25). But now that you mention it, I see that `k8s.io/apimachinery/pkg/util/yaml` is only used once in `provider/github` and the `gopkg.in/yaml.v2` dependency is used more often.

Therfore, it might make sense to only use `gopkg.in/yaml.v2` so that we can ditch `k8s.io/apimachinery/pkg/util/yaml`. What do you think?

Reply:
@zakisk I switched from `k8s.io/apimachinery/pkg/util/yaml` to `gopkg.in/yaml.v2` for yaml validation.

Reply:
i think if i rem ok we need gopkg.yaml for yaml error validation in that pacakge, all other we use k8 machinery yaml because it remove a lot of extra stuff when reading/writting a yaml

i think the e2e test fails on that

Reply:
@tricktron can you please take at [this line](https://github.com/openshift-pipelines/pipelines-as-code/actions/runs/12177814138/job/33968434297?pr=1846#step:11:2024), this is the reason E2E is failing.